src/main/java/week2/BTree.java

```diff
@@ -0,0 +1,113 @@
+package week2;
+
+
+class BTree<T extends Comparable<T>> {
+
+  private boolean bothFound = false;
+  private Node<T> root;
```

Comment:
I don't like these variables. I don't see a particular reason to make the methods anything other than pure functions - you could just pass in the root, rather than constructing an object with the root set, and then call the method on the object.

I particularly don't like "bothFound", as it's internal state that's not obvious, and it prevents the code from being used in a multithreaded manner, despite effectively performing pure function/side-effect free work.